src/utils/permissions.ts

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import memoize from "nano-memoize";
+
+import { D2 } from "../types/d2";
+
+export enum AppRoles {
+    METADATA_SYNC_ADMINISTRATOR = "METADATA_SYNC_ADMINISTRATOR",
+}
+
+export const isUserAdmin = memoize(async (d2: D2) => {
+    const userRoles = await d2.currentUser.getUserRoles();
+    return userRoles
+        .toArray()
+        .find((role: any) => role.name === AppRoles.METADATA_SYNC_ADMINISTRATOR);
+});
+
+export const getUserInfo = memoize(
+    async (
+        d2: D2
+    ): Promise<{
+        userGroups: any[];
+        id: string;
+        name: string;
+        username: string;
+    }> => {
+        const userGroups = await d2.currentUser.getUserGroups();
+
+        return {
+            userGroups: userGroups.toArray(),
+            id: d2.currentUser.id,
+            name: d2.currentUser.name,
+            username: d2.currentUser.username,
+        };
+    }
+);
+
+export const initializeAppRoles = async (baseUrl: string) => {
+    for (const role in AppRoles) {
+        console.log(role, AppRoles);
+        const { userRoles } = (await axios.get(baseUrl + "/metadata", {
+            withCredentials: true,
+            params: {
+                userRoles: true,
+                filter: `name:eq:${role}`,
+                fields: "id",
+            },
+        })).data as { userRoles?: { id: string }[] };
+
+        if (!userRoles || userRoles.length === 0) {
+            await axios.post(
```

Comment:
Some validation maybe handy, nothing fancy, just checking status for example.

Reply:
This will only happen on first load of the application or if someone deleted the roles manually.

I wanted to control the scenario where the app has just been opened for the first time and the user does not have the authority to import metadata (which will likely fail with 200-ERROR).